server/filestore.go

```diff
@@ -4765,6 +4769,13 @@ func (fs *fileStore) selectMsgBlockWithIndex(seq uint64) (int, *msgBlock) {
 		if seq > last {
 			low = mid + 1
 		} else if seq < first {
+			// A message block's first sequence can change here meaning we could find a gap.
+			// We want to behave like above, which if inclusive (we check at start) should
+			// awlays return an index and a valid mb.
```

Comment:
```suggestion
			// always return an index and a valid mb.
```

Reply:
Thx will fix. I do run spellcheck but I guess its not catching much these days..